src/render/src/lib.rs

```diff
@@ -14,8 +14,63 @@
 
 #![deny(missing_docs)]
 
-//! An efficient, low-level, bindless graphics API for Rust. See [the
-//! blog](http://gfx-rs.github.io/) for explanations and annotated examples.
+//! # gfx
+//!
+//! An efficient, low-level, bindless graphics API for Rust.
+//!
+//! # Overview
+//!
+//! ## Command buffers and encoders
+//!
+//! A command buffer is an already built serialized list drawing and compute commands.
+//! Unlike with vulkan, command buffers are what you use to create commands, but only
+//! the result of creating these commands. Gfx, borrowing metal's terminology, uses
+//! encoders to build command buffers. This means that in general users of the gfx crate
+//! don't manipulate command buffers directly much and interact mostly with encoders.
+//!
+//! Manipulating an `Encoder` in gfx corresponds to interacting with:
+//!
+//! - a `VkCommandBuffer` in vulkan,
+//! - a `MTLCommandEncoder` in metal,
+//! - an `ID3D12GraphicsCommandList` in D3D12.
+//!
+//! OpenGL and earlier versions of D3D don't have an explicit notion of command buffers
+//! or encoders (with the exception of draw indirect commands in late versions of OpenGL,
+//! which can be seen as a GPU-side command buffer). They are managed implicitly by the driver.
+//!
+//! See:
+//!
+//! - The [`Encoder` struct documentation](struct.Encoder.html).
+//! - The [`Command buffer` trait documentation](trait.CommandBuffer.html).
+//!
+//! ## Factory
+//!
+//! The factory is what lets you allocate GPU resources such as buffers and textures.
+//!
+//! Each gfx backend provides its own factory type which implements both:
+//!
+//! - The [`factory` trait documentation](traits/trait.Factory.html#overview).
+//! - The [`factoryExt` trait documentation](traits/trait.FactoryExt.html).
```

Comment:
"documentation" seems out of place here. Also, traits should be capitalized correctly.